src/neo/SmartContract/ExecutionContextState.cs

```diff
@@ -11,5 +11,15 @@ internal class ExecutionContextState
         /// Calling script hash
         /// </summary>
         public UInt160 CallingScriptHash { get; set; }
+
+        /// <summary>
+        /// Allow call another contract
+        /// </summary>
+        public bool AllowCall { get; set; } = true;
+
+        /// <summary>
+        /// Allow to modify the state, False is the same as ReadOnly mode
+        /// </summary>
+        public bool AllowModifyStates { get; set; } = true;
```

Comment:
AllowStorageChanges?

Which other state is able to be modified? ABI? If so, maybe maybe to keep the current name.

Reply:
AllowWrites?